src/inmanta/ast/statements/generator.py

```diff
@@ -387,10 +439,47 @@ def get_attributes(self) -> Dict[str, ExpressionStatement]:
         """
         return self.__attributes
 
+    def get_wrapped_kwargs(self) -> List["WrappedKwargs"]:
+        """
+            Get the wrapped kwargs that are set for this constructor call
+        """
+        return self.__wrapped_kwarg_attributes
+
     attributes = property(get_attributes)
+    wrapped_kwargs = property(get_wrapped_kwargs)
 
     def __repr__(self) -> str:
         """
             The representation of the this statement
         """
         return "Construct(%s)" % (self.class_type)
+
+
+class WrappedKwargs(ExpressionStatement):
+    """
+    Keyword arguments wrapped in a dictionary
```

Comment:
add a comment that this AST node is there only for the typecheck.

Kwargs might have been a better name than WrappedKwargs

Reply:
I started out with Kwargs but then I realized that the normal way of passing arguments is using kwargs already. These are kwargs wrapped in a dict, therefore kwargs. I am willing to change it to Kwargs though if you believe it will be benificial.
What do you mean with this AST node is there only for the typecheck?